resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/LocalDirsFeatureStepSuite.scala

```diff
@@ -116,4 +115,33 @@ class LocalDirsFeatureStepSuite extends SparkFunSuite {
         .withValue(defaultLocalDir)
         .build())
   }
+
+  test("local dir on mounted volume") {
+    val volumeConf = KubernetesVolumeSpec(
+      "spark-local-dir-test",
+      "/tmp",
+      "",
+      false,
+      KubernetesHostPathVolumeConf("/hostPath/tmp")
+    )
+    val mountVolumeConf = KubernetesTestConf.createDriverConf(volumes = Seq(volumeConf))
+    val mountVolumeStep = new MountVolumesFeatureStep(mountVolumeConf)
+    val configuredPod = mountVolumeStep.configurePod(SparkPod.initialPod())
+
+    val sparkConf = new SparkConfWithEnv(Map())
+    val localDirConf = KubernetesTestConf.createDriverConf(sparkConf)
```

Comment:
You can use the same `KubernetesConf` for all steps right? That's how it works when spark-submit is run.

Reply:
That is the idea, to keep building out one conf.  (There were some early proposals around making these stages "more functional" but those were considered less broadly familiar and less spark idiomatic)